test/web3/Web3Spec/Live/MultifilterSpec.purs

```diff
@@ -0,0 +1,102 @@
+                  Api.eth_sendTransaction $ txOpts # _data ?~ MultifilterCode.deployBytecode
+                                                   # _value ?~ (mkValue zero :: Value Wei)
+              ) $
+      it "can receive multiple events in the correct order" \contractCfg -> do
+        let {contractAddress: multifilterAddress, userAddress} = contractCfg
+            txOpts = defaultTestTxOptions
+                       # _from ?~ userAddress
+                       # _to ?~ multifilterAddress
+            vals1 = 1..5
+            vals2 = 6..10
+            nVals = length vals1 + length vals2
+            fireE1 n = void $ assertWeb3 provider $ Multifilter.fireE1 txOpts {_value : mkUIntN s256 n}
+            fireE2 n = void $ assertWeb3 provider $ Multifilter.fireE2 txOpts {_value: mkUIntN s256 n}
+
+            filter1 = eventFilter (Proxy :: Proxy Multifilter.E1) multifilterAddress
+            filter2 = eventFilter (Proxy :: Proxy Multifilter.E2) multifilterAddress
+
+        raceV <- AVar.new []
+
+        count1V <- AVar.new 0
+        f1 <- forkWeb3 provider $ 
+          let h1 = mkHandler (Proxy :: Proxy Multifilter.E1) provider raceV count1V (_ == length vals1) true
+          in event filter1 h1
+
+        count2V <- AVar.new 0
+        f2 <- forkWeb3 provider $ 
+          let h2 = mkHandler (Proxy :: Proxy Multifilter.E2) provider raceV count2V (_ == length vals2) false
+          in event filter2 h2
+
+        syncV <- AVar.new []
+        sharedCountV <- AVar.new 0
+        let multiFilter = { e1 : filter1, e2 : filter2}
+            h1 = mkHandler (Proxy :: Proxy Multifilter.E1) provider syncV sharedCountV (_ == nVals) true
+            h2 = mkHandler (Proxy :: Proxy Multifilter.E2) provider syncV sharedCountV (_ == nVals) false
+            multiHandler = {e1: h1, e2: h2}
+        f3 <- forkWeb3 provider $ event' multiFilter multiHandler {trailBy: 0, windowSize: 0}
+        parSequence_ $
+          [ parTraverse_ fireE1 vals1
+          , parTraverse_ fireE2 vals2
+          ]
+        _ <- joinWeb3Fork f1
+        _ <- joinWeb3Fork f2
+        _ <- joinWeb3Fork f3
+        
+        race <- AVar.take raceV
+        sync <- AVar.take syncV
+        
+        race `shouldNotEqual` sync
+        sort race `shouldEqual` sync
+        sort sync `shouldEqual` sync
+
+mkHandler
+  :: forall e.
+     Proxy e
+  -> Provider
+  -> AVar.AVar (Array (Tuple BlockNumber BigNumber))
+  -> AVar.AVar Int
+  -> (Int -> Boolean)
+  -> Boolean
+  -> EventHandler Web3 e
+mkHandler _ provider indexV countV p shouldDelay = \e -> do
+  Change c <- ask
+  iAcc <- liftAff $ AVar.take indexV
+  let index = Tuple c.blockNumber c.logIndex
+      indices = iAcc `snoc` index
+  liftAff $ AVar.put indices indexV
+  count <- liftAff $ AVar.take countV
+  let count' = count + 1
+  liftAff $ AVar.put count' countV
+  if p count'
+    then pure TerminateEvent
+    else do
+      when shouldDelay $ 
```

Comment:
what's the point of this shouldDelay?

Reply:
the point is to ensure that the `E1` handler is running much slower than the `E2` handler, which when they are running separately means that `E2`'s handler is shoving in event incides into their shared `AVar` much faster than `E1`. 

This way in the non-multifilter case you are more certain that they are  processed out of order, even though the test explicitly checks this in order to even validate the test https://github.com/f-o-a-m/purescript-web3/pull/126/files#diff-66b441c78c7ed1f688999fcd6dea5a76R75

In the synchronous case this delay does not effect the order, it just means that `E2`s handler has to wait for the delay as well.